public/javascripts/common/MissionTutorial.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * Provides structure for a slide based tutorial framework.
+ *
+ * @param labelTypeModule
+ * This object contains the following:
+ *     - missionInstruction1: Text to be shown at the very top, above the slides area.
+ *     - missionInstruction2: Text to be shown below missionInstruction1, above the slides area.
+ *     - slides: An array of 'slides'.
+ *
+ *     Each 'slide' contains the following:
+ *         - isExampleCorrect: boolean, indicating whether the example type is correct or incorrect.
+ *         - exampleTypeLabel: string, text indicating whether the example type is correct or incorrect.
+ *         - exampleTypeIcon: string, ID of the SVG icon to be shown. A symbol with this ID should be present
+ *                            in icons.scala.html.
+ *         - slideTitle: string, title for the slide.
+ *         - slideSubtitle: string, subtitle for the slide.
+ *         - slideDescription: string, long form text description for the slide.
+ *         - imageURL: string, URL to the image to be shown.
+ *         - onImageLabel: object, containing the following:
+ *             - title: string, title for the on-image label
+ *             - description: string, description for the on-image label.
+ *             - position: object, containing 'top' and 'left' attributes for the on-image label.
+ *                         The 'top' and 'left' must be defined with respect to the top and left
+ *                         of the 'image element.'
+ *
+ * @constructor
+ */
+function MissionTutorial(labelTypeModule) {
```

Comment:
Even though it's longer, maybe we rename stuff that says "mission tutorial" to "mission start tutorial"? The added clarity sounds worth it to me here.

Reply:
Sure.